### PR TITLE
Add documentation for static events/queue

### PR DESCRIPTION
**Pull request by maciejbocianski**
This PR contains documentation update to cover new features of  EventQueue and UserAllocatedEvent classes (PR with the implementation https://github.com/ARMmbed/mbed-os/pull/11342)

PR with code example https://github.com/ARMmbed/mbed-os-examples-docs_only/pull/52

### CODE REVIEW

**Review by AnotherButler — COMMENTED** [review]
Please review my changes to make sure I didn't accidentally change the meaning of anything.

**Comment by maciejbocianski** [comment]
Updated example link

**Comment by maciejbocianski** [comment]
@AnotherButler 
LGTM

File: docs/api/rtos/EventQueue.md [hunk 1]
```diff
@@ -57,6 +57,12 @@ To do this, set the `mbed_app.json` configuration option `events.shared-dispatch
 
 [![View code](https://www.mbed.com/embed/?url=https://os.mbed.com/teams/mbed_example/code/Shared_Events_2/)](https://os.mbed.com/teams/mbed_example/code/Shared_Events_2/file/154179bdc39d/main.cpp)
 
+## Static EventQueue example: posting user allocated events to the static queue
+
+Use static EventQueue to prevent your program failing due to queue memory exhaustion or to prevent dynamic memory allocation:
```
Comment (jamesbeyond) [inline docs/api/rtos/EventQueue.md]: shouldn't it be ` prevent your program from failing` ?

File: docs/api/rtos/UserAllocatedEvent.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+- `call_on` and `try_call_on` to bind and post an event to the EventQueue as a function argument.
+- `cancel` to cancel the most recently posted event.
+
+Because the `UserAllocatedEvent` holds event data, you can post only one event object to it at a time. This means that if the event object has to be reused, the previous dispatch has to finish or the event has to be canceled. You can use the `try_call` API to sample the event state. This call tries to post an event and returns false with no action until the previous dispatching finishes.
+
+The UserAllocatedEvent class is thread safe. The `call`, `try_call` and `cancel` APIs are IRQ safe.
+
+## UserAllocatedEvent class reference
+
+[![View code](https://www.mbed.com/embed/?type=library)](https://os.mbed.com/docs/development/mbed-os-api-doxy/_event_8h_source.html)
```
Comment (jamesbeyond) [inline docs/api/rtos/UserAllocatedEvent.md]: Why this link still goes to the `Event` class in doxygen, should this be to `UserAllocatedEvent` class ?

**Comment by AnotherButler** [comment]
@jamesbeyond Good catches 👍 Does that look better?

**Review by jamesbeyond — APPROVED** [review]
Thanks! that looks good to me!

**Comment by AnotherButler** [comment]
As soon as the example is approved and merged, I'll merge this PR.

**Review by AnotherButler — APPROVED** [review]

**Pull request merged**